gstcefsrc.cc

```diff
@@ -177,7 +213,7 @@ class AudioHandler : public CefAudioHandler
     GstStructure *s = gst_structure_new ("cef-audio-stream-start",
         "channels", G_TYPE_INT, channels,
         "rate", G_TYPE_INT, params.sample_rate,
-        NULL);
+        nullptr);
```

Comment:
oh clang doesn't accept NULL as a sentinel ? that's dumb :facepalm:

Reply:
C++ has various type complications, and you should really use `nullptr` instead of `NULL`. `NULL` is an integer 0 in C++ (so you might or might not get a pointer-sized 0 here with the varargs depending on a lot of stuff), while `nullptr` is a pointer

Reply:
glib defines `NULL` as `nullptr` when C++11 is being used:

```c
#ifndef NULL
#  if G_CXX_STD_CHECK_VERSION (11)
#    define NULL (nullptr)
#  elif defined (G_CXX_STD_VERSION)
#    define NULL (0L)
#  else
#    define NULL ((void*) 0)
#  endif /* G_CXX_STD_CHECK_VERSION (11) */
#endif
```

Perhaps this doesn't work correctly on clang, or something else is defining `NULL` here.

Reply:
You can't rely on GLib's definition to be used. Every library has code like this, and until C++11 the correct definition of `NULL` was indeed `0L` (reason: `void *` does not implicitly convert to any other pointer type with C++ but a 0 integer does :clown_face: )

Reply:
oof OK, thanks for the explanation :)